FifthLesson/VectorProgram/VectorProgram/Model/Vector.cs

```diff
@@ -1,115 +1,137 @@
         {
-            return $"({CoordinateFirst}, {CoordinateSecond}, {CoordinateThird})";
+            return $"({FirstCoordinate}, {SecondCoordinate}, {ThirdCoordinate})";
         }
 
-        public static Vector operator +(Vector vector1, Vector vector2)
+        public override int GetHashCode() => GetHashCodeHelper(
+               new int[]
+               {
+                   FirstCoordinate.GetHashCode(),
+                   SecondCoordinate.GetHashCode(),
+                   ThirdCoordinate.GetHashCode()
+               }
+            );
+
+        public override bool Equals(object obj) => this.Equals((Vector)obj);
+        #endregion
+
+        #region operators + - * == !=
+        public static Vector operator +(Vector first, Vector second)
         {
-            var result = new Vector(vector1);
+            var result = new Vector(first);
 
-            result.CoordinateFirst += vector2.CoordinateFirst;
-            result.CoordinateSecond += vector2.CoordinateSecond;
-            result.CoordinateThird += vector2.CoordinateThird;
+            result.FirstCoordinate += second.FirstCoordinate;
+            result.SecondCoordinate += second.SecondCoordinate;
+            result.ThirdCoordinate += second.ThirdCoordinate;
 
             return result;
         }
 
-        public static Vector operator -(Vector vector1, Vector vector2)
+        public static Vector operator -(Vector first, Vector second)
         {
-            var result = new Vector(vector1);
+            var result = new Vector(first);
 
-            result.CoordinateFirst -= vector2.CoordinateFirst;
-            result.CoordinateSecond -= vector2.CoordinateSecond;
-            result.CoordinateThird -= vector2.CoordinateThird;
+            result.FirstCoordinate -= second.FirstCoordinate;
+            result.SecondCoordinate -= second.SecondCoordinate;
+            result.ThirdCoordinate -= second.ThirdCoordinate;
 
             return result;
         }
 
         public static Vector operator *(double numeric, Vector vector)
         {
-            return new Vector(vector.CoordinateFirst * numeric, vector.CoordinateSecond * numeric, vector.CoordinateThird * numeric);
+            return new Vector(vector.FirstCoordinate * numeric, vector.SecondCoordinate * numeric, vector.ThirdCoordinate * numeric);
         }
 
         public static Vector operator *(Vector vector, double numeric)
         {
             return numeric * vector;
         }
 
-        public static Vector operator *(Vector vector1, Vector vector2)
+        public static Vector operator *(Vector first, Vector second)
         {
-            return new Vector(vector1.CoordinateFirst * vector2.CoordinateFirst, vector1.CoordinateSecond * vector2.CoordinateSecond, vector1.CoordinateThird * vector2.CoordinateThird);
+            return new Vector(first.FirstCoordinate * second.FirstCoordinate, first.SecondCoordinate * second.SecondCoordinate, first.ThirdCoordinate * second.ThirdCoordinate);
```

Comment:
Use object initializer

Reply:
Corrected. The method was incorrectly implemented.
public static **double** operator *(Vector first, Vector second)